### PR TITLE
leofs: Detect if Application is up issue leo-project/leofs#852

### PR DESCRIPTION
**Pull request by windkit**
## Description
Use nodetool to check the list of running application, when the application is up, the component is ready

## Related to issue
#852

### CODE REVIEW

**Comment by mocchira** [comment]
@windkit Now the startup script of each component (manager, storage, gateway) is generated(copied when rebar generate) based on https://github.com/leo-project/leofs/blob/master/rel/common/launch.sh a common startup script file so you'd have to modify this file instead of rel/${component}/files/${component} files.
also nodetool seems to be redundant so it's time to move nodetool under rel/common and modify only this file to make the code less redundant and easy to maintain.

**Comment by windkit** [comment]
@mocchira OK I can then update the PR to consolidate them and remove the unnecessary files from git

**Comment by windkit** [comment]
@mocchira I have updated the PR, merged nodetool into one and updated the launch script.

We also have redundant files under `apps/leo_{gateway,storage,manager}/rel`, should we also link the files up the source tree?

**Comment by mocchira** [comment]
@windkit 

> I have updated the PR, merged nodetool into one and updated the launch script.

Thanks, I'll check it.

> We also have redundant files under apps/leo_{gateway,storage,manager}/rel, should we also link the files up the source tree?
Merge 

As other files seem to be remained without any modification for 5 yeas, keep those as is this time.

File: rel/common/nodetool [hunk 1]
```diff
@@ -76,6 +79,30 @@ main(Args) ->
...
+                                            ok;
+                                       (_, Acc) ->
+                                            Acc
+                                    end, waiting, List) of
+                        ok ->
+                            io:format("Application is ready\n");
+                        _ ->
+                            io:format("Application is not ready\n"),
+                            halt(1)
+                    end
```
Comment (mocchira) [inline rel/common/nodetool]: http://erlang.org/doc/man/lists.html#any-2 would be more suitable one as it stops to iterate when Predictor func returns true.

```erlang
case lists:any(
fun({leo_gateway,_,_}) ->
    true;
fun({leo_manager,_,_}) ->
    true;
fun({leo_storage,_,_}) ->
    true;
fun(_) ->
    false
end, List) of ->
    true ->
    false ->
end

**Comment by mocchira** [comment]
LGTM. Thanks for your contribution.

**Pull request merged**